src/background/vpncontroller/vpncontroller.js

```diff
@@ -218,23 +218,24 @@ export class VPNController extends Component {
   /**
    * Handles a response from the native messaging brige
    * @param {BridgeResponse} response - The Reponse object from the NM Bridge
-   * @param {VPNState} state - the current state
-   * @returns - Nothing, but may set state, or post messages to the bridge.
+   * @param {WritableProperty<VPNState>} state - the current state
+   * @returns - Nothing, but may write to state, or post messages to the bridge.
    */
-  async handleBridgeResponse(response, state = this.#mState.value) {
+  async handleBridgeResponse(response, state) {
+    const currentState = state.value;
     // We can only get 2 types of messages right now: client-down/up
     if (
       (response.status && response.status === "vpn-client-down") ||
       (response.error && response.error === "vpn-client-down")
     ) {
       // If we have been considering the client open, it is now closed.
-      if (state.alive) {
-        state = new StateVPNClosed();
```

Comment:
This is the function arg, not the actual property. sooo nothing was set.

Reply:
![](https://i.giphy.com/media/v1.Y2lkPTc5MGI3NjExcDRha2Zsc3pld2llb2k3cjl2N3dyOTAwcmlwbW92c2k5YWg1MXk1bCZlcD12MV9pbnRlcm5hbF9naWZfYnlfaWQmY3Q9Zw/ySpxjJmsq9gsw/giphy.gif)